tests/Microsoft.DotNet.Framework.Docker.Tests/ImageTests.cs

```diff
@@ -75,20 +77,33 @@ private static string GetFilterRegexPattern(string filter)
         }
 
         [Theory]
+        [Trait("Category", "Runtime")]
         [MemberData(nameof(GetVerifyImagesData))]
         public void VerifyImagesWithApps(ImageDescriptor imageDescriptor)
         {
-            VerifyImages(imageDescriptor, "dotnetapp", "", true);
+            VerifyFxImages(imageDescriptor, "dotnetapp", "", true);
         }
 
         [Theory]
+        [Trait("Category", "Runtime")]
         [MemberData(nameof(GetVerifyImagesData))]
         public void VerifyImagesWithWebApps(ImageDescriptor imageDescriptor)
         {
-            VerifyImages(imageDescriptor, "webapp", "powershell -command \"dir ./bin/SimpleWebApplication.dll\"", false);
+            VerifyFxImages(imageDescriptor, "webapp", "powershell -command \"dir ./bin/SimpleWebApplication.dll\"", false);
+        }
+
+        [Theory]
+        [Trait("Category", "WCF")]
+        [MemberData(nameof(GetVerifyImagesData))]
+        public void VerifyWcfImagesWithApps(ImageDescriptor imageDescriptor)
+        {
+            if (imageDescriptor.RuntimeVersion != "3.5")
```

Comment:
The `ImageDescriptor` that is passed in here is essentially a matrix consisting of runtime/SDK/OS values.  The test gets executed for each combination that is defined in the `TestData` field.  The problem with this is that the WCF tests do not even care about the SDK version.  The WCF tests are only using the runtime/OS values.  So the WCF tests end up processing multiple `ImageDescriptors` that consist of the same values for runtime/OS which amounts to running a duplicate test for no reason.

For example, there exists this snippet from the `TestData` field:
```
new ImageDescriptor { RuntimeVersion = "4.7.2", BuildVersion = "4.7.2", OsVariant = WSC_LTSC2019 },
new ImageDescriptor { RuntimeVersion = "4.7.2", BuildVersion = "4.8", OsVariant = WSC_LTSC2019 },
```

The only variation between those is the BuildVersion property.  The WCF tests don't consume this property  This means the WCF tests run a test for `RuntimeVersion = 4.7.2` and `OS = WSC_LTSC2019` twice!

To fix this, I think WCF should have its own set of ImageDescriptors that define the matrix of values it cares about, separate from the existing tests.

Reply:
will be fixed in the next commit